kubernetes_platform/proto/kubernetes_executor_config.proto

```diff
@@ -49,17 +49,20 @@ message EnabledSharedMemory {
 }
 
 message SecretAsVolume {
-    // Name of the Secret.
-    string secret_name = 1;
+    // Deprecated, use secret_name_parameter instead.
+    string secret_name = 1 [deprecated = true];

```

Comment:
I'm curious if it would be backwards compatible to use "one of" instead so that you can accept a string or `InputParameterSpec` but keep the same field name.

Reply:
if I'm understanding you correctly, this won't work because you can't have duplicate parameter names in a oneOf
 (I agree that the new name is non-ideal, I would prefer to keep the older name, open to suggestions here)
 
feel free to post a sample of what you mean and I can take a look!

Reply:
Oh right, but it still may be better to use `oneOf` if possible to avoid the user providing both values if it's backwards compatible.

Reply:
`oneof` isn't used for proto field migration purpose. When we recognize something needs migration, the correct pattern is to add new proto field and deprecate the old one.